app/src/main/java/nerd/tuxmobil/fahrplan/congress/utils/ConfirmationDialog.kt

```diff
@@ -0,0 +1,77 @@
+package nerd.tuxmobil.fahrplan.congress.utils
+
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+
+import androidx.annotation.CallSuper
+import androidx.annotation.MainThread
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import nerd.tuxmobil.fahrplan.congress.R
+
+class ConfirmationDialog : DialogFragment() {
+
+    interface OnConfirmationDialogClicked {
+        fun onAccepted(dlgRequestCode: Int)
+    }
+
+    companion object {
+        const val BUNDLE_DLG_TITLE = "ConfirmationDialog.DLG_TITLE"
+        const val BUNDLE_DLG_REQUEST_CODE = "ConfirmationDialog.DLG_REQUEST_CODE"
+        const val FRAGMENT_TAG = "ConfirmationDialog.FRAGMENT_TAG"
+
+        fun newInstance(@StringRes dlgTitle: Int, requestCode: Int): ConfirmationDialog {
+            val dialog = ConfirmationDialog()
+
+            dialog.listener = null
+
+            val args = Bundle()
+            args.putInt(BUNDLE_DLG_TITLE, dlgTitle)
+            args.putInt(BUNDLE_DLG_REQUEST_CODE, requestCode)
+            dialog.arguments = args
+
+            dialog.isCancelable = false
+
+            return dialog
+        }
+    }
+
+    private var dlgTitle = 0
+    private var dlgRequestCode = 0
+    private var listener: OnConfirmationDialogClicked? = null
+
+    @MainThread
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val args = requireArguments()
+        dlgTitle = args.getInt(BUNDLE_DLG_TITLE)
+        dlgRequestCode = args.getInt(BUNDLE_DLG_REQUEST_CODE)
+    }
+
+    @MainThread
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
```

Comment:
There is some potential to make this more Kotlin-ish. Do you see the **Lint warnings** in Android Studio? You can refactor the function to this:

``` kotlin
@MainThread
override fun onCreateDialog(savedInstanceState: Bundle?) = AlertDialog
    .Builder(requireContext())
    .setTitle(title)
    .setPositiveButton(R.string.dlg_delete_all_favorites_delete_all) { _, _ ->
        listener?.onAccepted(requestCode)
    }
    .setNegativeButton(android.R.string.cancel) { _, _ ->
        // Do nothing.
    }
    .create()
```

Reply:
Actually, it showed that warning, but I wasn't aware that you could name multiple parameters to `_` and though Android studio would then try to rename the second parameter to `_1` or something, which would just make the code less readable 😅 Thanks for pointing this out :)

Reply:
Please have a look closely - I did more than just the `_`.